stretch-assignment/index.js

```diff
@@ -1,9 +1,6 @@
-let topBlock = document.querySelectorAll('.block'); 
+let topBlock = document.querySelectorAll('.block');
 
-let topBlockArray = Array.from(topBlock); 
+let topArray = Array.from(topBlock); 
 
-topBlockArray.forEach(function(block) {
-    block.addEventListener('click', () => topBlock.prepend());
+topArray.forEach(block => block.addEventListener('click', function() {block.parentNode.prepend(block)}));
```

Comment:
nice combination of variables, grabbing the elements and then turning them into an array to allow you to map through and prepend when a block is clicked on